syfertext/token.py

```diff
@@ -39,10 +44,69 @@ def __str__(self):
     def set_attribute(self, name: str, value: object):
         """Creates a custom attribute with the name `name` and
            value `value` in the Underscore object `self._`
+
+        Args:
+            name (str): name of the custom attribute.
+            value (object): value of the custom named attribute.
         """
 
+        # make sure that the name is not empty and does not contains any spaces
+        assert (
+            isinstance(name, str) and len(name) > 0 and (" " not in name)
+        ), "Argument name should be a non-empty str type containing no spaces"
+
         setattr(self._, name, value)
 
+    def has_attribute(self, name: str):
+        """Checks if the Underscore object `self._` has an attribute `name`
+
+        Args:
+            name (str): name of the custom attribute.
+        
+        Returns:
+            attr_exists (bool): `True` if `self._.name` exists, otherwise `False`  
+        """
+
+        # make sure that the name is not empty and does not contains any spaces
+        assert (
+            isinstance(name, str) and len(name) > 0 and (not (" " in name))
+        ), "Argument name should be a non-empty str type containing no spaces"
+
+        attr_exists = hasattr(self._, name)
+
+        return attr_exists
+
+    def remove_attribute(self, name: str):
+        """Removes the attribute `name` from the Underscore object `self._`
+
+        Args:
+            name (str): name of the custom attribute.
+        """
+
+        # Before removing it, it should exist
+        assert self.has_attribute(name), "token does not have the attribute {}".format(name)
+
+        delattr(self._, name)
+
+    def nbor(self, i=1):
+        """Gets the neighbouring token at `self.i + i` if it exists
```

Comment:
again type hint for return value  `def nbor(self, i=1) -> Token`

Reply:
please call `i` -> `offset`